### PR TITLE
Add sample scatter gather flow control

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
Add sample "scatter-gather flow control"

### CODE REVIEW

**Comment by abeykoon** [comment]
Please add descriptions to mediator elements. 

image

**Comment by abeykoon** [comment]
In data-mapper, i do not see any mapping done. 
Could you please check. 

image

File: migration/mule/scatter-gather-flow-control/ScatterGatherFlowControl/src/main/synapse-config/sequences/aggregate_seq.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<sequence name="aggregate_seq" trace="disable" xmlns="http://ws.apache.org/ns/synapse">
+    <property name="aggregated" scope="default">
+        <aggregated xmlns=""/>
+    </property>
+    <aggregate id="aggregate_id">
+        <completeCondition>
+            <messageCount max="-1" min="-1"/>
```
Comment (abeykoon) [inline migration/mule/scatter-gather-flow-control/ScatterGatherFlowControl/src/main/synapse-config/sequences/aggregate_seq.xml]: Does aggregate mediator detect that it has to wait for 2 clones?

File: migration/mule/scatter-gather-flow-control/ScatterGatherFlowControl/src/main/synapse-config/api/ScatterGatherFlowControl.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<api context="/aggregate" name="ScatterGatherFlowControl" xmlns="http://ws.apache.org/ns/synapse">
+    <resource methods="GET">
+        <inSequence>
+            <property action="remove" name="NO_ENTITY_BODY" scope="axis2"/>
+            <clone id="aggregate_id">
+                <target>
+                    <sequence>
+                        <fileconnector.read>
+                            <source>[file path of contacts-1.csv]</source>
```
Comment (abeykoon) [inline migration/mule/scatter-gather-flow-control/ScatterGatherFlowControl/src/main/synapse-config/api/ScatterGatherFlowControl.xml:10]: How do we skip reading the header line (first line in the .csv file)?

Comment (dulanjalidilmi) [reply migration/mule/scatter-gather-flow-control/ScatterGatherFlowControl/src/main/synapse-config/api/ScatterGatherFlowControl.xml:10]: This is fixed by providing numberOfLinesToSkip.

File: migration/mule/scatter-gather-flow-control/ScatterGatherFlowControl/src/main/synapse-config/sequences/aggregate_seq.xml (hunk 1 above)
Comment (dulanjalidilmi) [reply migration/mule/scatter-gather-flow-control/ScatterGatherFlowControl/src/main/synapse-config/sequences/aggregate_seq.xml]: Fixed.

**Comment by abeykoon** [comment]
@dulanjalidilmi Please raise an issue to track AI data mapping issue.

**Review by abeykoon — APPROVED** [review]

**Pull request merged**